### PR TITLE
 DBT-1107 Add pin field for captain adding fuel

### PR DESCRIPTION
**Pull request by kayiganwa04**
## Context

We are adding the Pin field to verify if it's the captain

## Test

- go to add fuel as a captain 

## Proof of Concept
image

### CODE REVIEW

File: src/components/fuel/index.tsx [hunk 1]
```diff
@@ -53,6 +57,13 @@ export default function FuelForm() {
   const submitFuelForm = async () => {
     setLoading(true);
     try {
+      const { Pin } = await getCaptain(data.Captain) as Captain
+      const CaptainPin = Pin.replace(/\n/g, "");
+
+      if (CaptainPin !== data.Pin) {
+        return toast.error(t("error.error_captain_pin"))
+      }
```
Comment (abelbordonado) [inline src/components/fuel/index.tsx:65]: This check needs to be done in the API, the PIN can not reach the client (this is a security concern)

File: src/components/common/inputs/input.tsx [hunk 2]
```diff
@@ -26,6 +27,7 @@ export default function CommonInput({
   max?: number;
   step?: number;
   checked?: boolean;
+  minlength?: any
```
Comment (abelbordonado) [inline src/components/common/inputs/input.tsx:30]: number I guess

File: src/components/fuel/index.tsx [hunk 3]
```diff
@@ -89,101 +100,142 @@ export default function FuelForm() {
...
+                      {showPin ?
+                        <div onClick={() => setShowPin(!showPin)} className="cursor-pointer">
+                          <svg className="h-6 text-gray-700" fill="none" xmlns="http://www.w3.org/2000/svg"
+                            viewBox="0 0 640 512">
+                            <path fill="currentColor"
+                              d="M320 400c-75.85 0-137.25-58.71-142.9-133.11L72.2 185.82c-13.79 17.3-26.48 35.59-36.72 55.59a32.35 32.35 0 0 0 0 29.19C89.71 376.41 197.07 448 320 448c26.91 0 52.87-4 77.89-10.46L346 397.39a144.13 144.13 0 0 1-26 2.61zm313.82 58.1l-110.55-85.44a331.25 331.25 0 0 0 81.25-102.07 32.35 32.35 0 0 0 0-29.19C550.29 135.59 442.93 64 320 64a308.15 308.15 0 0 0-147.32 37.7L45.46 3.37A16 16 0 0 0 23 6.18L3.37 31.45A16 16 0 0 0 6.18 53.9l588.36 454.73a16 16 0 0 0 22.46-2.81l19.64-25.27a16 16 0 0 0-2.82-22.45zm-183.72-142l-39.3-30.38A94.75 94.75 0 0 0 416 256a94.76 94.76 0 0 0-121.31-92.21A47.65 47.65 0 0 1 304 192a46.64 46.64 0 0 1-1.54 10l-73.61-56.89A142.31 142.31 0 0 1 320 112a143.92 143.92 0 0 1 144 144c0 21.63-5.29 41.79-13.9 60.11z">
+                            </path>
```
Comment (abelbordonado) [inline src/components/fuel/index.tsx:191]: SVG needs to be its own component.

And then this logic need to be in the SecretInput component

File: src/components/fuel/index.tsx [hunk 4]
```diff
@@ -89,101 +100,142 @@ export default function FuelForm() {
...
+                          </svg>
+                        </div>
+                        :
+                        <div onClick={() => setShowPin(!showPin)} className="cursor-pointer">
+                          <svg className="h-6 text-gray-700" fill="none" xmlns="http://www.w3.org/2000/svg"
+                            viewBox="0 0 576 512">
+                            <path fill="currentColor"
+                              d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z">
+                            </path>
+                          </svg>
```
Comment (abelbordonado) [inline src/components/fuel/index.tsx:201]: Own component

File: src/components/common/inputs/input.tsx (hunk 2 above)
Comment (kayiganwa04) [reply src/components/common/inputs/input.tsx:30]: Number is not working

File: README.md [hunk 5]
```diff
@@ -59,6 +59,7 @@ For the production was is generated as a system user in the facebook business se
 - PDFMONKEY_API_KEY: This API key you can get it from the PDF monkey in the account page or in notion env variables [PDF Monkey URL](https://dashboard.pdfmonkey.io/account)
 - MAKE_KEY: Is a key from make.com with permissions to run scenarios. Is used to run scenarios in demand (Like after generating the PDF)https://eu2.make.com/user/api
 
+- MASTER_PIN: Is a key which is going to be used by the super admin for some actions like sending the whatsapp messages, you can find it Notion
```
Comment (abelbordonado) [inline README.md:62]: Perfect ENV description

File: src/app/api/notion/notion.api.ts [hunk 6]
```diff
@@ -75,4 +75,23 @@ export async function createDatabaseItem({
     console.error("Create Item in Database :", error.message);
     return { error: error.message };
   }
-}
\ No newline at end of file
+}
+
+export async function removePrivateProperties(response: any) {
```
Comment (abelbordonado) [inline src/app/api/notion/notion.api.ts:80]: `any` => Record<sring,unknown>

File: src/app/api/verifyPin/route.ts [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import * as verify from "./verifyPin.api";
+
+export async function POST(request: Request) {
+  try {
+    // Ensure the Content-Type is 'application/json' for correct parsing
+    if (request.headers.get("Content-Type") !== "application/json") {
+      throw new Error("Invalid Content-Type");
+    }
+    const body = await request.json();
+    const captainId = body.captainId;
```
Comment (abelbordonado) [inline src/app/api/verifyPin/route.ts:10]: The backend doesnt know about captains, this is just itemId

and is not verifyCaptainPin, just verifyValue (for example)
tomorrow can be password 

Just a naming concern

File: src/app/api/verifyPin/verifyPin.api.ts [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import { Client } from "@notionhq/client";
+const notionSecret = process.env.NOTION_SECRET || undefined;
+const masterPin = process.env.MASTER_PIN;
+const notion = new Client({ auth: notionSecret });
+export async function verifyCaptainPin(captainId: string, Pin: string) {
+  try {
+    const response: any = await notion.pages.retrieve({
+      page_id: captainId,
+    });
+    const captainPin = (response.properties["$Pin"].rich_text[0].plain_text).replace(/\n/g, "")
```
Comment (abelbordonado) [inline src/app/api/verifyPin/verifyPin.api.ts:10]: You can not hardcode the `$Pin` here , this is something the client needs to say (body = {propertyName: "$Pin", value: "123123"

File: src/components/common/inputs/input.tsx (hunk 2 above)
Comment (abelbordonado) [reply src/components/common/inputs/input.tsx:30]: then string :D

File: src/app/api/verifyPin/verifyPin.api.ts (hunk 8 above)
Comment (kayiganwa04) [reply src/app/api/verifyPin/verifyPin.api.ts]: @abelbordonado what do you mean here?

**Pull request merged**